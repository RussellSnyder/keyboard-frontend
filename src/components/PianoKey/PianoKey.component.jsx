import React, { Component } from 'react';

const ATTACK_TIME = 20;

export class PianoKey extends Component {

    constructor(props) {
        super(props);
        this.handleMouseDown = this.handleMouseDown.bind(this);
        this.handleMouseUp = this.handleMouseUp.bind(this);
    }

    componentDidUpdate(prevProps) {
        if (prevProps.on && !this.props.on) {
            this.handleMouseUp()
        }
        if (!prevProps.on && this.props.on) {
            this.handleMouseDown()
        }
    }


    componentDidMount() {
    }

    handleMouseDown() {
        let {noteName} = this.props;
        this.props.playNoteHandler(noteName)
    }

    handleMouseUp() {
        let {noteName} = this.props;
        this.props.stopNoteHandler(noteName)
    }

    render() {
        let {points, className, noteName} = this.props;
        return (
            <polygon
                id={noteName}
                points={points}
                className={className + (this.props.on ? " playing":"")}
                onMouseDown={ this.handleMouseDown }
                onMouseUp={ this.handleMouseUp }
                onMouseOut={ this.handleMouseUp }
            />
        )
    }
}

export default PianoKey;
