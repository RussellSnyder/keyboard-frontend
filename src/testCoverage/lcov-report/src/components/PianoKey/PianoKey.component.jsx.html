<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/PianoKey/PianoKey.component.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/PianoKey</a> PianoKey.component.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.12% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/17</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-yes">73x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">75x</span>
<span class="cline-any cline-yes">75x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
&nbsp;
&nbsp;
export class PianoKey extends Component {
    constructor() {
        super();
        this.state = {
            active: false,
            hover: false
        }
        this.handleMouseDown = this.handleMouseDown.bind(this);
        this.handleMouseUp = this.handleMouseUp.bind(this);
    }
&nbsp;
    componentDidUpdate(prevProps, prevState, snapshot) {
        let {active} = this.props;
        <span class="missing-if-branch" title="if path not taken" >I</span>if (prevProps.active !== active) {
<span class="cstat-no" title="statement not covered" >            this.setState({active})</span>
        }
    }
&nbsp;
    componentDidMount() {
        let {active} = this.props;
        <span class="missing-if-branch" title="if path not taken" >I</span>if (active) <span class="cstat-no" title="statement not covered" >this.setState({active})</span>
    }
    handleMouseDown() {
        this.setState({ active: true })
        let {midiNote, noteName} = this.props;
        this.props.playNoteHandler(midiNote, noteName)
    }
&nbsp;
    handleMouseUp() {
        this.setState({ active: false })
        let {midiNote, noteName} = this.props;
        this.props.stopNoteHandler(midiNote, noteName)
    }
&nbsp;
    render() {
        let {points, className} = this.props;
        return (
            &lt;polygon
                points={points} className={className + (this.state.active ? " playing":"")}
                onMouseDown={ this.handleMouseDown }
                onMouseUp={ this.handleMouseUp }
            /&gt;
        )
    }
}
&nbsp;
export default PianoKey;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Sep 29 2018 22:51:29 GMT+0200 (W. Europe Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
