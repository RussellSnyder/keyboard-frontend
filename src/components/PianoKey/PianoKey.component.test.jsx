import React from 'react'
import renderer from 'react-test-renderer'

import {PianoKey} from "./PianoKey.component";
import { mount } from 'enzyme';

const stopNoteHandler = () => 'stop';
const playNoteHandler = () => 'play';

const component = renderer.create(<PianoKey
    stopNoteHandler={() => stopNoteHandler()}
    playNoteHandler={() => playNoteHandler()}
/>);

it('isRendered', () => {
    expect(component).toBeDefined();
});

it('Changes className when state active changes', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseDown();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseUp();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

