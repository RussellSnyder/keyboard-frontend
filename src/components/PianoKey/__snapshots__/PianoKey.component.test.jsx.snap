// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changes className when state active changes 1`] = `
<polygon
  className="undefined"
  onMouseDown={[Function]}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
/>
`;

exports[`Changes className when state active changes 2`] = `
<polygon
  className="undefined"
  onMouseDown={[Function]}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
/>
`;

exports[`Changes className when state active changes 3`] = `
<polygon
  className="undefined"
  onMouseDown={[Function]}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
/>
`;
